cosmoz-grouped-list.js

```diff
@@ -99,20 +100,93 @@
 		 */
 		_renderedItems: null,
 
+		/**
+		 * Instances of Template that are NOT available for reuse at the moment.
+		 */
+		_usedInstances: [],
+
+		/**
+		 *  Instances of Template that are available for reuse at the moment.
+		 */
+		_reusableInstances: [],
+
+		/**
+		 * Polymer `created` livecycle function.
+		 *
+		 * @return {void}
+		 */
 		created() {
 			this._templateSelectors = [];
 			this._renderedItems = [];
-			this._physicalItens = [];
 		},
 
+		/**
+		 * Polymer `attached` livecycle function.
+		 *
+		 * @return {void}
+		 */
 		attached() {
-			this._isAttached = true;
+			this._templatesObserver = Polymer.dom(this.$.templates)
+				.observeNodes(this._onTemplatesChange.bind(this));
 			this._debounceRender();
 		},
 
+		/**
+		 * Polymer `detached` livecycle function.
+		 *
+		 * @return {void}
+		 */
 		detached() {
-			this._isAttached = false;
+			if (this._templatesObserver) {
+				Polymer.dom(this).unobserveNodes(this._templatesObserver);
+				this._templatesObserver = null;
+			}
 			this.cancelDebouncer('render');
+
+
+			this._resetAllTemplates();
+
+			this._usedInstances = [];
+			this._reusableInstances = [];
+
+		},
+		_onTemplatesChange(change) {
+			if (!Array.isArray(this._ctors) || this._ctors.length === 0) {
+				const templates = Array.from(change.addedNodes)
+					.filter(n => n.nodeType === Node.ELEMENT_NODE && n.tagName === 'TEMPLATE' && n.matches('[class]'));
+
+				if (templates.length === 0) {
+					console.warn('cosmoz-grouped-list requires templates');
+					return;
+				}
+				const baseProps = {
+					[this.as]: true,
+					[this.indexAs]: true,
+					folded: true,
+					expanded: true,
+					selected: true,
+					highlighted: true
+				};
+
+				this._ctors = templates.reduce((ctors, template) => {
+					ctors[template.getAttribute('class').split(' ')[0]] = Cosmoz.Templatize.templatize(template, this, {
```

Comment:
Since you support multiple `class` values, then you shouldn't assume the expected class in in first position.
Something like 
```
const templateType = template.classList.contains('item') ? 'item' : 'group';
```
seems more appropriate

Reply:
Also, if we miss one of the template, we could write an error in the console, in order to help users.

Reply:
@plequang I have applied the requested changes.